### DISCUSSION TITLE
Are there any plans to create or be able to use connectionstyle attribute in Non-Directed graphs?

### DISCUSSION THREAD

**Discussion by jirp-upc** [networkx/networkx#5971] [Ideas]
Hi,

I'm coming from here:

[https://stackoverflow.com/questions/73583710/how-to-modify-position-or-placement-of-an-edge-in-graph-drawing](url)

There is any plan of make possible to use a connectionstyle attribute in Non-Directed graphs? I need to use them in order to do a proper representation of this kind of graphs developed in my project.

Thank you very much,

Ignacio

**Comment by dschult** [comment]
If you set the keyword argument `arrows =True` and `connectionstyle="whatever"` that should work.
The `arrows=True` is needed to indicate that a FancyArrowPatch is used instead of a LineCollection.

**Reply by jirp-upc** [reply]
Thanks. The fact is that I would be interested in set different connectionstyle attributes depending on the edge. Is it possible? When I do it like other personalized edge atributes, I get some errors.

For example, for simple bars, Im creating the edge assigning this parameter:

`G.add_edge(pt+j,pt+j+1,weight=1, color='black', connectionstyle="arc3,rad=0.0")`

And with curvature:

`G.add_edge(pt,(j+pt)-1,weight=1, color='black',connectionstyle="arc3,rad=0.3")`

I'm getting errors here:

` opt = { 'node_color': 'navajowhite',
        'node_size': 500,`
        'width': 3,
        'with_labels': True }
    coords =networkx.get_node_attributes(G,'pos')
    labels = networkx.get_edge_attributes(G,'weight')
    cs = networkx.get_edge_attributes(G,'connectionstyle')

    plt.close()
    networkx.draw(G, pos = coords, arrows=True, connectionstyle=cs, **opt) `

Last line throws this exception:
![image](https://user-images.githubusercontent.com/79309980/189119818-5804e75a-39de-403e-b15b-0caefc93fc14.png)

**Reply by jirp-upc** [reply]
I suppose that connectionstyle attribute expects a single string to put the same style in every edge of the graph, right? Is there a way to put specific connectionstyle depending on the edge?

**Comment by dschult** [comment]
You should use the "advanced" interface functions:
```python
nx.draw_networkx_nodes
nx.draw_networkx_labels
nx.draw_networkx_edges
```
The key is the last one which draws the edges.  That takes `edgelist` as an input. You can draw each edge one at a time with their own connectionstyle.  One of the pending PRs #5882 uses this technique [to draw multiedges with different arcs ](https://github.com/networkx/networkx/blob/4be1d1d8b86765de272e6015aed3823d37dd36b0/networkx/drawing/nx_pylab.py#L1028)

**Reply by jirp-upc** [reply]
Using nx.draw_network_edges:

`networkx.draw_networkx_edges(G, pos=coords,edgelist=[(node_1,node_2)], connectionstyle=G.edges[node_1][node_2]['connectionstyle'], arrows=True, width=3)`

How I can pass a single edge? Doing this throw another exception, 

Thank you very much.

Ignacio

**Reply by rossbar** [reply]
> How I can pass a single edge? Doing this throw another exception,

It's hard to know without a minimal example + the traceback. Here's an example of plotting different edges with different properties:

```python
import networkx as nx
G = nx.path_graph(3)
pos = {n: (n, n) for n in G}
nx.draw_networkx_nodes(G, pos)
nx.draw_networkx_edges(G, pos, edgelist=[(0, 1)])
nx.draw_networkx_edges(G, pos, edgelist=[(1, 2)], arrows=True, connectionstyle="arc3,rad=0.2")
```

which gives
![fig1](https://user-images.githubusercontent.com/1268991/189165859-c10dedc3-d43f-43fb-bab1-82ec8e16aec8.png)

In general I'd recommend you check out the [example gallery](https://networkx.org/documentation/latest/auto_examples/index.html) for visualization examples.

**Reply by jirp-upc** [reply]
Thank you very much, I am so noob with the use of the forums!! Lesson learned. 

I've used this code, I didn't realize that I was doing a bad use of code structuring:

`for e in G.edges():
        networkx.draw_networkx_edges(G, pos=coords,edgelist=[e], connectionstyle=G[e[0]][e[1]]['connectionstyle'], arrows=True, width=3)`

Thank you very much!!!!

Hope that in a future PRs mentioned above is solved,

Ignacio
